pinot-core/src/main/java/org/apache/pinot/core/operator/transform/function/PrestoDateTruncFunction.java

```diff
@@ -0,0 +1,262 @@
+ * specific language governing permissions and limitations
+ * under the License.
+ */
+package org.apache.pinot.core.operator.transform.function;
+
+import org.apache.pinot.core.common.DataSource;
+import org.apache.pinot.core.operator.blocks.ProjectionBlock;
+import org.apache.pinot.core.operator.transform.TransformResultMetadata;
+import org.apache.pinot.core.plan.DocIdSetPlanNode;
+import org.joda.time.Chronology;
+import org.joda.time.DateTimeField;
+import org.joda.time.DateTimeFieldType;
+import org.joda.time.DateTimeZone;
+import org.joda.time.DurationField;
+import org.joda.time.DurationFieldType;
+import org.joda.time.chrono.ISOChronology;
+import org.joda.time.field.DividedDateTimeField;
+import org.joda.time.field.OffsetDateTimeField;
+import org.joda.time.field.ScaledDurationField;
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+
+import javax.annotation.Nonnull;
+import java.util.List;
+import java.util.Map;
+import java.util.concurrent.TimeUnit;
+
+import static java.util.Locale.ENGLISH;
+import static org.apache.pinot.core.operator.transform.function.TimeZoneKey.MAX_TIME_ZONE_KEY;
+import static org.apache.pinot.core.operator.transform.function.TimeZoneKey.getTimeZoneKeys;
+
+/**
+ * The <code>PrestoDateTruncFunction</code> class implements the sql compatible date_trunc function for TIMESTAMP type.
+ * <p>
+ * <ul>
+ * <li>
+ * This transform function should be invoked with arguments:
+ * <ul>
+ * <li>
+ * Timestamp column input (or expression), this should always be in the specified units since epoch UTC
+ * </li>
+ * <li>
+ * Incoming units in terms of the TimeUnit enum. For example NANOSECONDS, SECONDS etc
+ * </li>
+ * <li>
+ * Time zone as specified in the zone-index.properties file, these timezones are both DateTime and JodaTime compatible
+ * </li>
+ * <li>
+ * Unit: second, minute, hour, day, week, month, quarter, year
+ * </li>
+ * </ul>
+ * </li>
+ * </ul>
+ * It mimics the Presto function "date_trunc(TIMESTAMP, 'unit') -> TIMESTAMP" specifically and provides support for multiple timestamps
+ * <p>
+ * This function is intended to be mainly used by presto when presto is translating queries from Presto/ANSI SQL to Pinot QL.
+ * <p>
+ * It exists because neither of the two existing UDF's, TimeConvert nor DateTimeConvert, provide SQL like compatibility, making this translation
+ * from presto date_trunc to Pinot impossible.
+ * <p>
+ * Example translations:
+ * <ul>
+ * <li>
+ * // Assume that the presto session timezone is 'Europe/Berlin'
+ * date_trunc(from_unixtime(time_in_seconds), 'week') to prestoDateTrunc(time_in_seconds, 'SECONDS', 'Europe/Berlin', 'week')
+ * <p>
+ * // timestamp with timezone (override whatever the default session timezone is)
+ * date_trunc(from_unixtime(time_in_seconds, 'Americas/New_York'), 'week') to prestoƒDateTrunc(time_in_seconds, 'SECONDS', 'Americas/New_York', 'week')
+ * </li>
+ * </ul>
+
+ */
+public class PrestoDateTruncFunction
```

Comment:
Is this the same as `PrestoDateTrunc.java`? I believe `PrestoDateTrunc.java` should be deleted.
+1 on removing `Presto` from the function name, and rename it to `DateTruncTransformFunction`